### PR TITLE
chore: refactor wds combobox component

### LINKED ISSUE #36224: Improve combobox styles

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
Added styles for input to the combobox
Separate reused components(FieldDescription, FieldError, FieldLabel, FieldListPopover)

Fixes #36224

## Automation

/ok-to-test tags="@tag.Anvil"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10888601720>
> Commit: 7ef81de082771d0345478f5cf21ce7fadcc83547
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10888601720&attempt=1" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.Anvil`
> Spec:
> <hr>Mon, 16 Sep 2024 17:33:23 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

## Release Notes

- **New Features**
  - Added a more comprehensive set of button size examples to the `Button.stories.tsx` file.
  - Introduced a new `FieldDescription` component in the `ComboBox.tsx` file to handle field descriptions.

- **Styling Improvements**
  - Refined the spacing, padding, and text size for the `Button` component based on its size.
  - Enhanced the visual feedback for checkboxes in invalid and selected states in the `Checkbox` component.
  - Simplified the CSS structure and improved the visual states of the `ComboBox` component.

- **Refactor**
  - Replaced legacy components in the `ComboBox` with more standardized design system components for improved consistency.
  - Removed the `ListBoxItem` component, indicating a change in the list box rendering approach.
  - Narrowed the allowed sizes for the `ComboBox` component, excluding "xSmall" and "large" sizes.

- **Chores**
  - Removed the `ErrorMessage` component and its associated export in the `index.ts` file.
  - Added a `verbatimModuleSyntax` property to the `tsconfig.json` file, potentially affecting module handling.

### CODE REVIEW

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css [hunk 1]
```diff
@@ -78,7 +78,7 @@
     margin-block-end: var(--body-margin-end);
     inline-size: var(--body-line-height);
     margin-inline-start: var(--body-margin-start);
-    margin-inline-end: var(--body-margin-end);
+    margin-inline-end: var(--body-margin-start);
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:81]: Fix margins for icons. cc @ichik

File: app/client/packages/design-system/widgets/src/components/Checkbox/src/styles.module.css [hunk 2]
```diff
@@ -96,11 +96,21 @@
     --checkbox-bg-color: var(--color-bd-negative);
   }
 
+  &[data-invalid][data-selected="true"] [data-icon] {
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/Checkbox/src/styles.module.css:99]: The colors weren't right here.  cc @ichik 
Снимок экрана 2024-09-11 в 17 31 56

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css (hunk 1 above)
Comment (ichik) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:81]: A bit weird semantically, but as long as this works. :)

Comment (jsartisan) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:81]: Do we have different values for `var(--body-margin-start)` and `var(--body-margin-end)`?

**Review by znamenskii-ilia — APPROVED** [review]

File: app/client/packages/design-system/widgets/src/components/ComboBox/src/ComboBox.tsx [hunk 3]
```diff
@@ -41,41 +34,23 @@ export const ComboBox = (props: ComboBoxProps) => {
...
-
-          {/* TODO: Use proper headless Input once Valera is back */}
           <div className={styles.inputWrapper}>
             <Input className={styles.input} placeholder={placeholder} />
-            <Button icon="chevron-down" isLoading={isLoading} size={size} />
+            <Button
+              icon="chevron-down"
+              isLoading={isLoading}
+              size={size}
+              slot={Boolean(isLoading) ? null : ""}
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/components/ComboBox/src/ComboBox.tsx:48]: What is slot?

**Review by jsartisan — APPROVED** [review]

File: app/client/packages/design-system/widgets/src/components/ComboBox/src/ComboBox.tsx (hunk 3 above)
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/ComboBox/src/ComboBox.tsx:48]: https://github.com/appsmithorg/appsmith/pull/34217#discussion_r1639578383

Specifically here, this code does not allow to open a dropdown if `isLoading` is true.

cc @znamenskii-ilia

File: app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css (hunk 1 above)
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/components/Button/src/styles.module.css:81]: Yep. We get it from capsize.

**Review by ichik — APPROVED** [review]
Visuals and main behavior is all good now. Two things will need a follow-up (one definitely, one is minor).

1. If you enter some value in the ComboBox and remove the focus that value is lost even if it fully matches one of the options. To make matters worse even if you press Enter after entering the value it doesn't actually select the value, you need to select the value with the arrow keys or mouse click and confirm it with Enter (confirming it with Space key doesn't work either). See the attached screen recording.
2. The inner dropdown button is not focusable with the keyboard. Ideally users should be able to focus it so that they can open the list of the available options.


https://github.com/user-attachments/assets/7933c706-8b22-4b7a-9b5b-b46ccfac33a8

**Review by ichik — APPROVED** [review]

**Pull request merged**